packages/mint-components/src/components/sqm-leaderboard/sqm-leaderboard-view.tsx

```diff
@@ -18,41 +22,156 @@ export interface LeaderboardViewProps {
   const { states, data, elements } = props;
   const { styles } = states;
-  if (states.loading) {
-    return elements.loadingstate;
-  }
+
+  if (states.loading)
+    return (
+      <div class={sheet.classes.Leaderboard}>
+        <style type="text/css">
+          {styleString}
+          {vanillaStyle}
+        </style>
+        {elements.loadingstate}
+      </div>
+    );
+
+  if (!states.hasLeaders) return elements.empty;
+
+  let userSeenFlag = false;
+
   return (
-    <div>
-      {!states.hasLeaders && elements.empty}
-      {states.hasLeaders && (
-        <table>
+    <div class={sheet.classes.Leaderboard}>
+      <style type="text/css">
+        {styleString}
+        {vanillaStyle}
+      </style>
+      <table>
+        <tr>
+          {styles.showRank && <th class="Rank">{styles.rankheading}</th>}
+          <th class="User">{styles.usersheading}</th>
+          <th class="Score">{styles.statsheading}</th>
+        </tr>
+        {data.leaderboard?.map((user) => {
+          if (user.rowNumber === data.viewerRank?.rowNumber)
+            userSeenFlag = true;
+          return (
+            <tr
+              class={
+                !styles.hideViewer &&
+                user.rowNumber === data.viewerRank?.rowNumber
+                  ? "highlight"
+                  : ""
+              }
+            >
+              {styles.showRank && <td class="Rank">{user.rank}</td>}
+              <td class="User">
+                {user.firstName && user.lastInitial
+                  ? user.firstName + " " + user.lastInitial
+                  : user.firstName || user.lastInitial
+                  ? user.firstName || user.lastInitial
+                  : styles.anonymousUser || "Anonymous User"}
+              </td>
+              <td class="Score">{user.value}</td>
+            </tr>
+          );
+        })}
+        {!userSeenFlag && !styles.hideViewer && (
           <tr>
-            {styles.showRank && <th class="Rank">{styles.rankheading}</th>}
-            <th class="User">{styles.usersheading}</th>
-            <th class="Score">{styles.statsheading}</th>
+            <td colSpan={100} class="ellipses">
+              <sl-icon
+                name="three-dots"
+                style={{ verticalAlign: "middle" }}
+              ></sl-icon>
+            </td>
+          </tr>
+        )}
+        {!userSeenFlag && !styles.hideViewer && (
+          <tr class="highlight">
+            {styles.showRank && (
+              <td class="Rank">{data.viewerRank?.rank || "-"}</td>
+            )}
+            <td class="User">
+              {data.viewerRank?.firstName && data.viewerRank?.lastInitial
+                ? data.viewerRank?.firstName +
+                  " " +
+                  data.viewerRank?.lastInitial
+                : data.viewerRank?.firstName || data.viewerRank?.lastInitial
+                ? data.viewerRank?.firstName || data.viewerRank?.lastInitial
+                : styles.anonymousUser || "Anonymous User"}
```

Comment:
If the prop has a default value, then why are we using a hardcoded fallback in the code?

Reply:
removed!